src/core/qfieldcloudprojectsmodel.cpp

```diff
@@ -359,262 +403,596 @@ void QFieldCloudProjectsModel::cancelDownloadProject( const QString &projectId )
+    QFieldCloudUtils::setProjectSetting( project->id, QStringLiteral( "canRepackage" ), project->canRepackage );
+    QFieldCloudUtils::setProjectSetting( project->id, QStringLiteral( "needsRepackaging" ), project->needsRepackaging );
+
+    emit dataChanged( projectIndex, projectIndex, QVector<int>::fromList( roleNames().keys() ) );
+    emit projectRefreshed( projectId, refreshReason );
+  } );
+}
+
+void QFieldCloudProjectsModel::projectStartJob( const QString &projectId, const JobType jobType )
+{
+  const QModelIndex projectIndex = findProjectIndex( projectId );
+  if ( !projectIndex.isValid() )
+    return;
+
+  CloudProject *project = mProjects[projectIndex.row()];
+  project->jobs[jobType] = Job( QString(), projectId, jobType );
 
   emit dataChanged( projectIndex, projectIndex, QVector<int>::fromList( roleNames().keys() ) );
 
-  QgsLogger::debug( QStringLiteral( "Packaging requested for \"%1\"" ).arg( projectId ) );
+  auto getStringJobType = []( JobType jobType ) {
+    switch ( jobType )
+    {
+      case JobType::Package:
+        return QStringLiteral( "package" );
+    }
 
-  // //////////
-  // 1) request a new project packaging
-  // //////////
-  /// If you want to skip packaging as being the slowest action, change /qfield-files/export/<PROJECT_ID> from POST to GET request
-  NetworkReply *reply = mCloudConnection->post( QStringLiteral( "/api/v1/qfield-files/export/%1/" ).arg( projectId ) );
-  project->apiNetworkReply = reply;
+    return QString();
+  };
 
-  connect( reply, &NetworkReply::finished, reply, [=]() {
-    QNetworkReply *rawReply = reply->reply();
+  QString jobTypeName = getStringJobType( jobType );
+  QgsLogger::debug( QStringLiteral( "Project %1: creating a new `%2` job..." ).arg( projectId, jobTypeName ) );
+  NetworkReply *reply = mCloudConnection->post(
+    QStringLiteral( "/api/v1/jobs/" ),
+    QVariantMap(
+      {
+        { "project_id", projectId },
+        { "type", jobTypeName },
+      } ) );
 
+  connect( reply, &NetworkReply::finished, reply, [=]() {
     reply->deleteLater();
-    project->apiNetworkReply = nullptr;
 
-    if ( rawReply->error() != QNetworkReply::NoError )
+    if ( project->isPackagingAborted )
     {
-      QgsLogger::debug( QStringLiteral( "Packaging request for \"%1\" finished with error: %2" ).arg( projectId, rawReply->error() ) );
-      projectDownloadFinishedWithError( projectId, QFieldCloudConnection::errorString( rawReply ) );
+      QgsLogger::debug( QStringLiteral( "Project %1: job creation finished, but project operations are aborted" ).arg( projectId ) );
       return;
     }
 
-    QgsLogger::debug( QStringLiteral( "Packaging request for \"%1\" finished with no error" ).arg( projectId ) );
+    // the project has been deleted
+    if ( !findProject( projectId ) )
+    {
+      QgsLogger::debug( QStringLiteral( "Project %1: job creation finished, but the project is deleted." ).arg( projectId ) );
+      return;
```

Comment:
Do you need to emit a projectJobFinished here with an error string or that's not needed? Same for the above if ( project->isPackagingAborted ) condition.

Reply:
I would say I missed them, thanks for spotting. It's would be huge improvement when half of these checks get unnecessary with dedicated `CloudProject` class.